<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>README.utf8.md</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Irene results</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">CLL</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>




</div>


<div id="irene-integrative-ranking-with-epigenetic-network-of-enhancers" class="section level1">
<h1>Irene: Integrative Ranking with Epigenetic Network of Enhancers</h1>
<p><a href="https://travis-ci.org/hdsu-bioquant/irene"><img src="https://travis-ci.org/hdsu-bioquant/irene.svg?branch=master" alt="Build Status" /></a></p>
<p><em>Irene</em> is an R package which allows you - Find significantly altered genes between two biological conditions from histone ChIP-Seq and DNA methylation tracks (BigWig) - Focus on the genes which are associated with more extensive epigenetic modification over targetting enhancers. - Annotate and visualize the global epigenetic variances with network analysis.<br />
Using <em>Irene</em>, we analyzed the epigenetic networks from multiple histone marks and DNA methylation in <a href="https://www.ncbi.nlm.nih.gov/geo/roadmap/epigenomics/">Roadmap</a>, <a href="http://dcc.blueprint-epigenome.eu/#/experiments">BLUEPRINT</a>, and <a href="http://www.epigenomes.ca/">CEEHRC</a>, and the results are presented in <a href="http://hdsu-bioquant.github.io/irene-web/" class="uri">http://hdsu-bioquant.github.io/irene-web/</a>.</p>
</div>
<div id="introduction" class="section level1">
<h1>INTRODUCTION</h1>
<p><em>Irene</em> is developed for two purposes in epigenetic ranking: - Integrate several epigenetic marks - Incorporate enhancers</p>
<p>With the help of <em>Irene</em>, user not only discover the genes which show significantly epigenetic alterations on their promoters, but also the ones which are connected with strong epigenetic modifications on neighbouring enhancers, which are presented as increased/decreased rankings (Fig. 1) <img src="https://raw.githubusercontent.com/hdsu-bioquant/irene-web/master/images/irene_fig.jpg" alt="example bipartite graph" /> <em>Fig. 1 Genes with more enhancer alterations have higher rank</em></p>
<p>The whole idea has been demonstrated in the Chapter III of Qi Wang’s <a href="https://github.com/hdsu-bioquant/irene-web/blob/master/docs/chapter3.pdf">dissertation</a>. For the above purposes, we employed singular value decomposition <a href="http://www.biostat.jhsph.edu/~hji/dpca/">dPCA</a> and random walk ranking <a href="http://igraph.org/r/doc/page_rank.html">PageRank</a>. The epigenetic alterations over genomic regulatory elements between two groups are presented as dPC scores and further to PageRank scores during solving the enhancer-promoter relationships.</p>
</div>
<div id="installation" class="section level1">
<h1>Installation</h1>
<pre class="sh"><code>tar --exclude=&#39;irene/.git&#39; --exclude=&#39;irene/res&#39; --exclude=&#39;irene/docs&#39; -zcvf irene.tar.gz irene</code></pre>
<p><em>Irene</em> can be installed directly from GitHub with the help of <em>devtools</em> package:</p>
<pre class="r"><code>library(devtools)
install_github(&quot;hdsu-bioquant/irene&quot;)</code></pre>
<p>If you have problems in intalling the package, please have a look at <a href="#faqs">FAQs</a>.</p>
</div>
<div id="for-impatient-people" class="section level1">
<h1>For impatient people</h1>
<p>If you do not have the various datasets needed in the following sections, please follow the <a href="#fullcode">supplymentary</a>, where we have prepared all the necessary input needed for performing the tests. In the test cases, you can understand the data structures, procedures, and the outputs of <em>Irene</em>.</p>
</div>
<div id="prerequisites" class="section level1">
<h1>Prerequisites</h1>
<div id="genomic-regions-to-be-tested" class="section level2">
<h2>Genomic regions to be tested</h2>
<p>User need to provide promoter and enhancer regions to compare their differential epigenetic modifications. We also provide pre-defined promoters from <a href="http://epd.vital-it.ch/index.php">The Eukaryotic Promoter Database</a> and pre-defined from <a href="http://www.genecards.org/Guide/GeneCard">GeneHancer</a>. Given that the enhancers from GeneHancer database are an ensemble of all tissues/cell types, one may need to filter out the unspecific ones, which can be done by overlapping with the enhancer-specific histone marks, e.g., histone H3 lysine 4 monomethylation (H3K4me1) or the histone 3 lysine 27 acetylation (H3K27ac) marks.</p>
</div>
<div id="promoter-enhancer-p-e-interactions" class="section level2">
<h2>Promoter-enhancer (P-E) interactions</h2>
<p>Enhancers within 1Mb distance to the transcription start site (TSS) are considered potential promoter-interacting ones. In a common sense, the interactions should not cross the topologically associating domains (TAD) boundary. Therefore, we compiled a cell-type specific enhancer-promoter interaction list by excluding the interactions outside the same TAD from <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE87112">GSE87112</a>. Enhancer-promoter interactions probability are estimated using a power-law decay function based on the distances to the TSS. Enhancer-promoter distances for sample tissues can be downloaded from <a href="https://github.com/hdsu-bioquant/irene-data/tree/master/PEdistances">https://github.com/hdsu-bioquant/irene-data/PEdistances</a>, including: - <a href="https://github.com/hdsu-bioquant/irene-data/blob/master/PEdistances/H1.hg19.pair.gz">H1</a>: H1 human embryonic stem cell line - <a href="https://github.com/hdsu-bioquant/irene-data/blob/master/PEdistances/MES.hg19.pair.gz">MES</a>: H1 BMP4 derived mesendoderm cultured cells - <a href="https://github.com/hdsu-bioquant/irene-data/blob/master/PEdistances/MSC.hg19.pair.gz">MSC</a>: H1 derived mesenchymal stem cells - <a href="https://github.com/hdsu-bioquant/irene-data/blob/master/PEdistances/NPC.hg19.pair.gz">NPC</a>: H1 derived neural precursor cells - <a href="https://github.com/hdsu-bioquant/irene-data/blob/master/PEdistances/TPC.hg19.pair.gz">TPC</a>: H1 derived trophoblast stem cells</p>
<p>In practice, as TADs between different cell types are relative conserved (<a href="#schmitt-ad-2016">Schmitt AD, 2016</a>), one can use the H1 cell line in case the TAD of corresponding cell type is not available.</p>
</div>
<div id="epigenetic-intensity-data" class="section level2">
<h2>Epigenetic intensity data</h2>
<em>Irene</em> requires user to provide <a href="https://genome.ucsc.edu/goldenpath/help/bigWig.html"><strong>BigWig</strong></a> format to represent the sequencing density of BS-Seq or ChIP-Seq. To begin with, user need to create a <strong>data.frame</strong> named <em>meta</em> indicating the location of the BigWig files, as well as groups and experiment types (as <em>dataset</em> column). <a id="meta">
<details>
<p><summary>Here is a sample as follows (click to uncollapse):</summary></p>
<table>
<colgroup>
<col width="3%" />
<col width="80%" />
<col width="6%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th></th>
<th>file</th>
<th>group</th>
<th>dataset</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>UCSD.H1_Derived_Mesenchymal_Stem_Cells.Bisulfite-Seq.methylC-seq_h1-msc_r1a.wig.bw</td>
<td>1</td>
<td>1</td>
</tr>
<tr class="even">
<td>2</td>
<td>UCSD.H1_Derived_Mesenchymal_Stem_Cells.Bisulfite-Seq.methylC-seq_h1-msc_r2a.wig.bw</td>
<td>1</td>
<td>1</td>
</tr>
<tr class="odd">
<td>3</td>
<td>UCSD.H1_Derived_Mesenchymal_Stem_Cells.H3K27ac.SK436.wig.bw</td>
<td>1</td>
<td>2</td>
</tr>
<tr class="even">
<td>4</td>
<td>UCSD.H1_Derived_Mesenchymal_Stem_Cells.H3K27ac.SK438.wig.bw</td>
<td>1</td>
<td>2</td>
</tr>
<tr class="odd">
<td>5</td>
<td>UCSD.H1_Derived_Mesenchymal_Stem_Cells.H3K27me3.SK437.wig.bw</td>
<td>1</td>
<td>3</td>
</tr>
<tr class="even">
<td>6</td>
<td>UCSD.H1_Derived_Mesenchymal_Stem_Cells.H3K27me3.SK439.wig.bw</td>
<td>1</td>
<td>3</td>
</tr>
<tr class="odd">
<td>7</td>
<td>UCSD.H1_Derived_Mesenchymal_Stem_Cells.H3K9ac.SK518.wig.bw</td>
<td>1</td>
<td>4</td>
</tr>
<tr class="even">
<td>8</td>
<td>UCSD.H1_Derived_Mesenchymal_Stem_Cells.H3K9ac.SK519.wig.bw</td>
<td>1</td>
<td>4</td>
</tr>
<tr class="odd">
<td>9</td>
<td>UCSD.H1_Derived_Mesenchymal_Stem_Cells.H3K9me3.SK507.wig.bw</td>
<td>1</td>
<td>5</td>
</tr>
<tr class="even">
<td>10</td>
<td>UCSD.H1_Derived_Mesenchymal_Stem_Cells.H3K9me3.SK508.wig.bw</td>
<td>1</td>
<td>5</td>
</tr>
<tr class="odd">
<td>11</td>
<td>UCSD.H1_Derived_Mesenchymal_Stem_Cells.Input.SK443.wig.bw</td>
<td>1</td>
<td>6</td>
</tr>
<tr class="even">
<td>12</td>
<td>UCSD.H1_Derived_Mesenchymal_Stem_Cells.Input.SK444.wig.bw</td>
<td>1</td>
<td>6</td>
</tr>
<tr class="odd">
<td>13</td>
<td>GSM429321_UCSD.H1.Bisulfite-Seq.methylC-seq_h1_r2a.bw</td>
<td>2</td>
<td>1</td>
</tr>
<tr class="even">
<td>14</td>
<td>GSM429322_UCSD.H1.Bisulfite-Seq.methylC-seq_h1_r2b.bw</td>
<td>2</td>
<td>1</td>
</tr>
<tr class="odd">
<td>15</td>
<td>GSM432685_UCSD.H1.Bisulfite-Seq.methylC-seq_h1_r1a.bw</td>
<td>2</td>
<td>1</td>
</tr>
<tr class="even">
<td>16</td>
<td>GSM432686_UCSD.H1.Bisulfite-Seq.methylC-seq_h1_r1b.bw</td>
<td>2</td>
<td>1</td>
</tr>
<tr class="odd">
<td>17</td>
<td>UCSD.H1.H3K27ac.LL313.bw</td>
<td>2</td>
<td>2</td>
</tr>
<tr class="even">
<td>18</td>
<td>UCSD.H1.H3K27ac.SAK270.bw</td>
<td>2</td>
<td>2</td>
</tr>
<tr class="odd">
<td>19</td>
<td>UCSD.H1.H3K27me3.LL241.bw</td>
<td>2</td>
<td>3</td>
</tr>
<tr class="even">
<td>20</td>
<td>UCSD.H1.H3K27me3.LL314.bw</td>
<td>2</td>
<td>3</td>
</tr>
<tr class="odd">
<td>21</td>
<td>UCSD.H1.H3K27me3.YL95.bw</td>
<td>2</td>
<td>3</td>
</tr>
<tr class="even">
<td>22</td>
<td>UCSD.H1.H3K9ac.LL240.wig.bw</td>
<td>2</td>
<td>4</td>
</tr>
<tr class="odd">
<td>23</td>
<td>UCSD.H1.H3K9ac.SAK68.wig.bw</td>
<td>2</td>
<td>4</td>
</tr>
<tr class="even">
<td>24</td>
<td>UCSD.H1.H3K9me3.AK54.wig.bw</td>
<td>2</td>
<td>5</td>
</tr>
<tr class="odd">
<td>25</td>
<td>UCSD.H1.H3K9me3.LL218.wig.bw</td>
<td>2</td>
<td>5</td>
</tr>
<tr class="even">
<td>26</td>
<td>UCSD.H1.H3K9me3.YL75.wig.bw</td>
<td>2</td>
<td>5</td>
</tr>
<tr class="odd">
<td>27</td>
<td>UCSD.H1.H3K9me3.YL77.wig.bw</td>
<td>2</td>
<td>5</td>
</tr>
<tr class="even">
<td>28</td>
<td>UCSD.H1.Input.AK57.wig.bw</td>
<td>2</td>
<td>6</td>
</tr>
<tr class="odd">
<td>29</td>
<td>UCSD.H1.Input.DM219.wig.bw</td>
<td>2</td>
<td>6</td>
</tr>
<tr class="even">
<td>30</td>
<td>UCSD.H1.Input.LL-H1-I1.wig.bw</td>
<td>2</td>
<td>6</td>
</tr>
<tr class="odd">
<td>31</td>
<td>UCSD.H1.Input.LL-H1-I2.wig.bw</td>
<td>2</td>
<td>6</td>
</tr>
<tr class="even">
<td>32</td>
<td>UCSD.H1.Input.LLH1U.wig.bw</td>
<td>2</td>
<td>6</td>
</tr>
<tr class="odd">
<td>33</td>
<td>UCSD.H1.Input.YL154.wig.bw</td>
<td>2</td>
<td>6</td>
</tr>
<tr class="even">
<td>34</td>
<td>UCSD.H1.Input.YL208.wig.bw</td>
<td>2</td>
<td>6</td>
</tr>
<tr class="odd">
<td>35</td>
<td>UCSD.H1.Input.YL262.wig.bw</td>
<td>2</td>
<td>6</td>
</tr>
<tr class="even">
<td>36</td>
<td>UCSD.H1.Input.YL328.wig.bw</td>
<td>2</td>
<td>6</td>
</tr>
</tbody>
</table>
</details>
<p>The <strong>BigWig</strong> files for our test cases were converted from wig files retrieved from <a href="https://www.ncbi.nlm.nih.gov/geo/roadmap/epigenomics/">NIH Roadmap Epigenomics Project data gateway</a>, <a href="http://dcc.blueprint-epigenome.eu/#/datasets">BLUEPRINT</a>, and <a href="http://www.epigenomes.ca/">CEEHRC</a>, respectively. Precompiled R objects in our test cases can be retrieved from <a href="#precompiled">here</a></p>
</div>
</div>
<div id="usage" class="section level1">
<h1>USAGE</h1>
<div id="r-environment-for-the-test-cases." class="section level2">
<h2>R environment for the test cases.</h2>
<p>The following options were set up during performing our tests.</p>
<pre class="r"><code>options(stringsAsFactors = FALSE)</code></pre>
</div>
<div id="load-pre-defined-regions" class="section level2">
<h2>Load pre-defined regions</h2>
<p>The pre-defined promoter and enhancer regions with corresponding IDs (precompiled <a href="https://github.com/hdsu-bioquant/irene-data/blob/master/promenh.hg19.bed">hg19</a>, <a href="https://github.com/hdsu-bioquant/irene-data/blob/master/promenh.hg38.bed">hg38</a>) need to be loaded first with:</p>
<pre class="r"><code>bed &lt;- read.table(&#39;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/promenh.hg19.bed&#39;)</code></pre>
</div>
<div id="import-sequencing-density-data-from-bigwig-files" class="section level2">
<h2>Import sequencing density data from BigWig files</h2>
<p>The datasets are prepared from BigWig files by users. User needs to create a <em>data.frame</em> named <strong>meta</strong> first using the BigWig files under current working directory, and <em>Irene</em> uses the <em><a href="http://hgdownload.soe.ucsc.edu/admin/exe/">bigWigAverageOverBed</a></em> program as follows, which can process BigWig files with multi-threads.</p>
<pre class="r"><code>data &lt;- importBW(meta, bed)</code></pre>
<p>Alternatively, users can prepare the same R data objects with cluster computing, following the sample <a href="#bigwigaverageoverbed">procedure</a> by themselves. <em>Irene</em> reads the data from <em>bigWigAverageOverBed</em> outputs into another <em>data.frame</em> named <strong>data</strong> over the input genomic regions. Afterwards, user needs to edit <strong>meta</strong> manually, which must contain <em>file, group, dataset</em> information (a sample <em>meta</em> is <a href="#meta">here</a>)</p>
</div>
<div id="filter-out-unspecific-enhancers" class="section level2">
<h2>Filter out unspecific enhancers</h2>
<p>We only took the regions which are more likely to be true enhancers, therefore the following function is used to get the indices which overlapped with enhancer histone marks (H3K4me1 in the following case). The peaks identified by Roadmap Epigenetics Project were retrieved from <a href="http://egg2.wustl.edu/roadmap/data/byFileType/peaks/consolidated" class="uri">http://egg2.wustl.edu/roadmap/data/byFileType/peaks/consolidated</a>, and run:</p>
<pre class="r"><code>i &lt;- filterPeak(c(&quot;GSM3444631.bed&quot;,&quot;GSM3444640.bed&quot;,&quot;GSM3444652.bed&quot;,&quot;GSM3444655.bed&quot;,
    &quot;GSM3444653.bed&quot;,&quot;GSM3444654.bed&quot;,&quot;GSM3444656.bed&quot;), bed, group=c(1,1,1,1,2,2,2))</code></pre>
<p><a id="precompiled"><strong>Below are the precompiled R data files in our test cases (see following) contain all the above mentioned objects (<em>meta, bed, data, i</em>) for the following tests</strong>.</p>
<table>
<thead>
<tr class="header">
<th>Cancer / primary cells</th>
<th>Controls</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Chronic Lymphocytic Leukemia (<a href="https://github.com/hdsu-bioquant/irene-data/blob/master/CLL.hg19.rda">CLL</a>)</td>
<td>Naive B Cell</td>
</tr>
<tr class="even">
<td>Acute Lymphoblastic Leukaemia (<a href="https://github.com/hdsu-bioquant/irene-data/blob/master/ALL.hg38.rda">ALL</a>)</td>
<td>Naive B Cell</td>
</tr>
<tr class="odd">
<td>Acute Myeloid Leukaemia (<a href="https://github.com/hdsu-bioquant/irene-data/blob/master/nkAML.hg38.rda">AML</a>)</td>
<td>Naive B Cell</td>
</tr>
<tr class="even">
<td>Multiple Myeloma (<a href="https://github.com/hdsu-bioquant/irene-data/blob/master/MM.hg38.rda">MM</a>)</td>
<td>Naive B Cell</td>
</tr>
<tr class="odd">
<td>Mantle Cell Lymphoma (<a href="https://github.com/hdsu-bioquant/irene-data/blob/master/MCL.hg38.rda">MCL</a>)</td>
<td>Naive B Cell</td>
</tr>
<tr class="even">
<td>Chronic Lymphocytic Leukemia (mutated) (<a href="https://github.com/hdsu-bioquant/irene-data/blob/master/mCLL.hg38.rda">mCLL</a>)</td>
<td>Naive B Cell</td>
</tr>
<tr class="odd">
<td>Colorectal Cancer (<a href="https://github.com/hdsu-bioquant/irene-data/blob/master/CRC.hg19.rda">CRC</a>)</td>
<td>Sigmoid Colon</td>
</tr>
<tr class="even">
<td>Lower Grade Glioma (<a href="https://github.com/hdsu-bioquant/irene-data/blob/master/GLM.hg19.rda">LGG</a>)</td>
<td>Normal Brain</td>
</tr>
<tr class="odd">
<td>Papillary Thyroid Cancer (<a href="https://github.com/hdsu-bioquant/irene-data/blob/master/PTC.hg19.rda">PTC</a>)</td>
<td>Normal Thyroid</td>
</tr>
<tr class="even">
<td>Mesenchymal Stem Cells (<a href="https://github.com/hdsu-bioquant/irene-data/blob/master/MSC.hg19.rda">MSC</a>)</td>
<td>Embryonic Stem Cells</td>
</tr>
<tr class="odd">
<td>Neural Progenitor Cells (<a href="https://github.com/hdsu-bioquant/irene-data/blob/master/NPC.hg19.rda">NPC</a>)</td>
<td>Embryonic Stem Cells</td>
</tr>
<tr class="even">
<td>Trophoblast Stem Cells (<a href="https://github.com/hdsu-bioquant/irene-data/blob/master/TSC.hg19.rda">TSC</a>)</td>
<td>Embryonic Stem Cells</td>
</tr>
<tr class="odd">
<td>H1 BMP4 derived Mesendoderm (<a href="https://github.com/hdsu-bioquant/irene-data/blob/master/MES.hg19.rda">MES</a>)</td>
<td>Embryonic Stem Cells</td>
</tr>
<tr class="even">
<td>Glioblastoma (GB) subtype <a href="https://github.com/hdsu-bioquant/irene-data/blob/master/GSE121719-raw.hg19.rda">IDH</a></td>
<td>GB subtype MES</td>
</tr>
<tr class="odd">
<td>Glioblastoma (GB) subtype <a href="https://github.com/hdsu-bioquant/irene-data/blob/master/GSE121719-raw.hg19.rda">RTK I</a></td>
<td>GB subtype MES</td>
</tr>
<tr class="even">
<td>Glioblastoma (GB) subtype <a href="https://github.com/hdsu-bioquant/irene-data/blob/master/GSE121719-raw.hg19.rda">RTK II</a></td>
<td>GB subtype MES</td>
</tr>
</tbody>
</table>
<p><em>Full R code for analyzing the above datasets is in the <a href="#fullcode">supplymentary</a>. For the CLL test case, which is already incorporated in the package, one can simply load necessary dataset with</em>:</p>
<pre class="r"><code>data(CLL)</code></pre>
</div>
<div id="measure-combinatorial-effect-of-epigenetic-alterations" class="section level2">
<h2>Measure combinatorial effect of epigenetic alterations</h2>
<p>We use <a href="http://www.biostat.jhsph.edu/~hji/dpca/">dPCA</a>(<a href="#ji-h-2013">Ji H, 2013</a>) to measure combinatorial effect of epigenetic alterations. The software is already integrated into <em>Irene</em> as an external C function. User can select a subset of datasets to study, for the CLL test case:</p>
<pre class="r"><code>j &lt;- c(1,2,6,8)</code></pre>
<p>Use the following command to read the data which were selected with index <em>i</em>:</p>
<pre class="r"><code>res &lt;- dPCA(meta, bed[i,], data[i,], datasets=j, transform=j, normlen=j, verbose=TRUE)</code></pre>
<p>or the following if all the genomic regions are to be tested.</p>
<pre class="r"><code>res &lt;- dPCA(meta, bed, data, datasets=j, transform=j, normlen=j, verbose=TRUE)</code></pre>
<p>The output <em>res</em> is a named list which has three keys: <em>gr</em>, <em>Dobs</em>, and <em>proj</em>.</p>
<ul>
<li><strong>gr</strong> contains the pre-defined regions followed by the computed PCs from dPCA. A sample output may look like this:</li>
</ul>
<table>
<thead>
<tr class="header">
<th>seqnames</th>
<th>start</th>
<th>end</th>
<th>id</th>
<th>PC1</th>
<th>PC2</th>
<th>PC3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>chr5</td>
<td>159099769</td>
<td>159099829</td>
<td>EBF1_1</td>
<td>22.12935</td>
<td>-11.913720</td>
<td>-1.9786781</td>
</tr>
<tr class="even">
<td>chr19</td>
<td>53869401</td>
<td>53869461</td>
<td>MYADM_3</td>
<td>22.12694</td>
<td>-8.876568</td>
<td>3.7468742</td>
</tr>
<tr class="odd">
<td>chr4</td>
<td>52862267</td>
<td>52862327</td>
<td>RASL11B_1</td>
<td>21.89510</td>
<td>-14.516893</td>
<td>-4.8862062</td>
</tr>
<tr class="even">
<td>chr8</td>
<td>141099562</td>
<td>141099711</td>
<td>GH08I141099</td>
<td>21.88550</td>
<td>7.578159</td>
<td>1.3920079</td>
</tr>
<tr class="odd">
<td>chr6</td>
<td>131063357</td>
<td>131063417</td>
<td>EPB41L2_4</td>
<td>21.57229</td>
<td>-12.886749</td>
<td>-0.8408109</td>
</tr>
<tr class="even">
<td>chr2</td>
<td>158457054</td>
<td>158457114</td>
<td>PKP4_1</td>
<td>21.42755</td>
<td>-8.421447</td>
<td>-4.6387676</td>
</tr>
</tbody>
</table>
<p>The table is sorted descendingly by PC1</p>
<ul>
<li><p><strong>Dobs</strong> The D matrix, which contains the observed differences between the two conditions. This is the data analyzed by dPCA.</p></li>
<li><strong>proj</strong> Estimated beta coefficients for each dPC. The contribution of each histone mark to the PCs can be visualized with the following function:</li>
</ul>
<pre class="r"><code>plotD(res$Dobs, c(&quot;K4me1&quot;, &quot;K4me3&quot;, &quot;K27ac&quot;, &quot;Meth&quot;), scales::percent(res$proj[&#39;percent_var&#39;,]))</code></pre>
</div>
<div id="infer-promoter-enhancer-interaction-probabilities-according-to-distances" class="section level2">
<h2>Infer promoter-enhancer interaction probabilities according to distances</h2>
<p>Use the following function to convert the promoter-enhancer interactions in a given range of TSS (1Mb in the data provided) to probabilities of interaction.</p>
<ul>
<li>We first load the P-E interactions in H1ESC cell line, the three columns in <em>H1</em> data.frame are enhancer IDs, promoter IDs, P-E distances, respectively.</li>
</ul>
<pre class="r"><code>H1 &lt;- read.table(&#39;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/PEdistances/H1.hg19.pair&#39;)</code></pre>
<ul>
<li>Then transform the P-E interactions from bp to Mb:</li>
</ul>
<pre class="r"><code>H1[,3] &lt;- abs(H1[,3]/1e6)</code></pre>
<ul>
<li>We apply a power-decay function to represent the likelihood of P-E interactions. The power-coefficient is estimated from several capture Hi-C datasets <a href="https://www.ebi.ac.uk/arrayexpress/experiments/E-MTAB-2323/">Mifsud et al, 2015</a>, <a href="https://www.ebi.ac.uk/ega/studies/EGAS00001001911">Javierre et al, ‎2016</a>, <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE84660">Rubin et al, 2017</a>, which is explained in Figure 3.11 and Figure 3.14b of Qi Wang’s <a href="https://github.com/hdsu-bioquant/irene-web/blob/master/docs/chapter3.pdf">dissertation</a>. We recommand using -20 for all the test cases.</li>
</ul>
<pre class="r"><code>H1[,3] &lt;- exp(-20*H1[,3]+1)</code></pre>
</div>
<div id="using-pagerank-to-rank-the-epigenetic-alterations-from-both-enhancers-and-promoters." class="section level2">
<h2>Using PageRank to rank the epigenetic alterations from both enhancers and promoters.</h2>
<p>The PageRank function will take in the alteration contribution from enhancers, resulting in higher ranks of associated promoters if targeted by highly altered enhancers.</p>
<pre class="r"><code>res$pg &lt;- pageRank(res$gr, H1)</code></pre>
<p>, where **res<span class="math inline">\(pg** contains rankings generated from PC1 to PC*n*. If one is interested in the second PC, it can be accessed by: ```r res\)</span>pg$PC2 ```</p>
</div>
<div id="get-the-rank-of-only-promoters-as-well-which-will-be-used-to-compare-with-the-pagerank-rankings." class="section level2">
<h2>Get the rank of only promoters as well, which will be used to compare with the PageRank rankings.</h2>
<p>Use the following function to get the rank of the promoters according to PC1. A higher rank indicates stronger alterations of the corresponding gene.</p>
<pre class="r"><code>res$pg$prom &lt;- getPromId(res$gr, pc=&quot;PC1&quot;)</code></pre>
</div>
<div id="use-literature-derived-marker-genes-as-a-metric-for-evaluating-the-ranking" class="section level2">
<h2>Use literature-derived marker genes as a metric for evaluating the ranking</h2>
<p>Cancer and cell-type specific marker genes are listed in Table S2 &amp; 3 of Qi Wang’s <a href="https://github.com/hdsu-bioquant/irene-web/blob/master/docs/chapter3.pdf">disseration</a>, which are already compiled as an R object and can be loaded with:</p>
<pre class="r"><code>data(markers)</code></pre>
<p>The Empirical Cumulative Distribution Function (ECDF) of the marker gene positions in each ranking list can be plotted with:</p>
<pre class="r"><code>plotRank(res$pg, markers$CLL)</code></pre>
<p>, where the CLL marker genes for this test are ordered according to the positions of their rankings (Fig. 2), and the area under the curve (AUC) of each ranking list is described in the legend.</p>
<div class="figure">
<img src="https://raw.githubusercontent.com/hdsu-bioquant/irene-web/master/images/irene_roc.jpg" alt="cll roc fig" />
<p class="caption">cll roc fig</p>
</div>
<p><em>Fig. 2 CLL marker gene positions along the ROC curves</em></p>
</div>
<div id="network-analysis-of-the-enriched-pathways-of-significantly-epigeneticaly-alterated-genes" class="section level2">
<h2>Network analysis of the enriched pathways of significantly epigeneticaly alterated genes</h2>
<p>Network analysis groups highly-ranked genes according to known gene interaction databases, and further searches clusters of genes in biological function databases. In this case, we loaded Human Protein Reference Database (<a href="http://www.hprd.org/">HPRD</a>) for grouping genes:</p>
<pre class="r"><code>data(hprd)</code></pre>
<p>, and set the edge weights in accordance with the average rank of the two connected genes, which is done by:</p>
<pre class="r"><code>g &lt;- edgeRank(res$pg$PC1, hprd)</code></pre>
<p>, where we created a weighted HPRD networks using the edge weights calculated from the PC1 of the genes.</p>
<p>Using random walk clustering, we got a list of top-ranked sub-networks with desired number, e.g., 15 sub-networks:</p>
<pre class="r"><code>res$gs &lt;- exportMultinets(g, 15)</code></pre>
<p>, and the sub-networks were searched in WikiPathways and KEGG for enrichment of biological functions:</p>
<pre class="r"><code>res$ga &lt;- annotNets(res$gs)</code></pre>
</div>
</div>
<div id="outputs" class="section level1">
<h1>Outputs</h1>
<p>The following commands need to be executed to generate HTML outputs for interactive exploration of the enriched networks, epigenetic tracks, rank comparisons and benchmarking (ROC curves). To begin with, a prefix variable should be set corresponding to the experiment first, as for this test case we set:</p>
<pre class="r"><code>prefix = &quot;CLL&quot;</code></pre>
<p>The files are created in the current working directory if not mentioned explicitly. Use the following commands to create a specific folder for the outputs (e.g., a sub-directory in the home folder):</p>
<pre class="r"><code>dir.create(&quot;~/CLLoutput&quot;)
setwd(&quot;~/CLLoutput&quot;)</code></pre>
<p>Write the two ranking list to a CSV file for comparison, significance of the genes from the highest to the lowest are ordered from the top to the bottom of the list.</p>
<pre class="r"><code>writeRank(res$pg[[1]], res$pg$prom, prefix)</code></pre>
<p>Write the normalized epigenome signals of the two groups, labels of the two groups are provided by user.</p>
<pre class="r"><code>writeData(res$gr, c(&quot;CLL&quot;, &quot;Bcell&quot;), prefix)</code></pre>
<p>Export the principal components of epigenetic marks</p>
<pre class="r"><code>exportD(res$Dobs, c(&#39;K4me1&#39;,&#39;K4me3&#39;,&#39;K27ac&#39;,&#39;Meth&#39;), &quot;CLL&quot;)</code></pre>
<p>, and the sub-networks list was saved to a JSON with each network structure:</p>
<pre class="r"><code>exportJSONnets(res$gs, prefix)</code></pre>
<p>In addition, the enriched pathways of the sub-networks are also saved as JSON file:</p>
<pre class="r"><code>exportJSONpathways(res$ga, prefix, n=15)</code></pre>
<p>Export the meta info to a CSV file if you have not done that before.</p>
<pre class="r"><code>write.csv(meta,paste0(prefix,&quot;meta.csv&quot;))</code></pre>
<p>Finally, create an index page in the same directory with above JSON files for visualizing in an internet browser:</p>
<pre class="r"><code>exportApps(prefix, markers[&#39;CLL&#39;])</code></pre>
</div>
<div id="interpreting-the-results" class="section level1">
<h1>Interpreting the results</h1>
<p>The above mentioned test cases are hosted in another repository <a href="http://hdsu-bioquant.github.io/irene-web" class="uri">http://hdsu-bioquant.github.io/irene-web</a>, in which the outputs are presented in the following four sections:</p>
<div id="rank-list" class="section level2">
<h2>Rank list</h2>
<p>For every test case, Irene produced two rank lists: one is from PageRank score (generated from promoter and enhancer scores, as well as P-E interactions, therefore abbreviated as PromEnh), another is from the PC1 scores of promoters (abbreviated as PromOnly). Both lists are contained in a CSV file for downloading, which looks like:</p>
<table>
<thead>
<tr class="header">
<th></th>
<th>PromEnh</th>
<th>PromOnly</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>MED13L</td>
<td>CBWD5</td>
</tr>
<tr class="even">
<td>2</td>
<td>CBLB</td>
<td>CBWD3</td>
</tr>
<tr class="odd">
<td>3</td>
<td>VAV3</td>
<td>CBWD7</td>
</tr>
<tr class="even">
<td>4</td>
<td>LPP</td>
<td>SERF1B</td>
</tr>
<tr class="odd">
<td>5</td>
<td>NOTCH2</td>
<td>SERF1A</td>
</tr>
<tr class="even">
<td>6</td>
<td>MKLN1</td>
<td>HIST2H4A</td>
</tr>
<tr class="odd">
<td>7</td>
<td>ARHGAP15</td>
<td>HIST2H4B</td>
</tr>
<tr class="even">
<td>8</td>
<td>BACH2</td>
<td>GTF2H2</td>
</tr>
<tr class="odd">
<td>9</td>
<td>MGAT5</td>
<td>NOTCH4</td>
</tr>
</tbody>
</table>
<p>…</p>
<p>, where the genes are ordered from the most significant to the least significant in both lists, and the line numbers correspond to the positions in the lists.</p>
<p>Plotting the positions of each gene in a two-dimensional <a href="http://hdsu-bioquant.github.io/irene-web/rank.html?sample=CLL">graph</a> allows one to inspect the level of significance regarding the promoters or enhancers of a gene, whereas the genes under high enhancer/low promoter regulation are placed at the bottom-right corner of the graph. In the above <em>CLL</em> test case, the result implies many genes are ranked higher considering the enhancer alterations. The baseline became distorted for those genes who did not receive contribution from enhancers. During benchmarking, The selected marker genes, as well as genes from other collections (<a href="https://cancer.sanger.ac.uk/cosmic">COSMIC</a>, <a href="https://www.malacards.org/">MalaCards</a>, <a href="https://www.intogen.org/">IntOGen</a>) are listed besides the graph, so that the genes are highlighted in red upon clicking the corresponding item. Additional information are shown as mouse-over tooltips or links in the pop-ups. <a href="https://hdsu-bioquant.github.io/irene-web/docs/Ranking_lists.pdf"><img src="https://raw.githubusercontent.com/hdsu-bioquant/irene-web/master/images/Ranking_lists.gif" alt="Click to see instructions on how to use this ranking plot" /></a> <em>Fig. 3 Exploring ranking plot interface (<a href="https://hdsu-bioquant.github.io/irene-web/docs/Ranking_lists.pdf">user guide</a>)</em></p>
</div>
<div id="network-enrichment" class="section level2">
<h2>Network enrichment</h2>
<p>In these network <a href="http://hdsu-bioquant.github.io/irene-web/net.html?sample=CLL">presentations</a>, the enriched sub-networks are ordered based on the average significance of the their nodes, wherein the left-side of the node represents its rank in the <em>PromEnh</em> list, and the right-side of the node represents its rank in the <em>PromOnly</em> list. Pathways of each sub-network are tested with <em>EnrichR</em>, and the ones which are significantly enriched in KEGG and WikiPathways are listed. Upon clicking an item in the drop list of pathways, the corresponding genes in that pathway are highlighted in red. Additional information are shown as mouse-over tooltips or links in the pop-ups.</p>
<p><a href="https://hdsu-bioquant.github.io/irene-web/docs/Network_enrichment.pdf"><img src="https://raw.githubusercontent.com/hdsu-bioquant/irene-web/master/images/Network_enrichment.gif" alt="Click to see instructions on how to use this network browser" /></a></p>
<p><em>Fig. 4 Exploring network browser interface (<a href="https://hdsu-bioquant.github.io/irene-web/docs/Network_enrichment.pdf">user guide</a>)</em></p>
</div>
<div id="epigenome-heatmap" class="section level2">
<h2>Epigenome heatmap</h2>
<p>The epigenome <a href="http://hdsu-bioquant.github.io/irene-web/browse.html?sample=CLL">heatmap</a> serves as an interactive browser of normalized epigenetic signals, wherein tracks from different samples are aligned in accordance with biological conditions and epigenetic marks. In this presentation, the whole genome is divided into bins with fixed length (2kb in this test case). To preserve storage space, only the bins overlaped with the provided promoter/enhancer regions are rendered, leaving the rest of the genome in background color. The topmost track represent the regulatory genomic elements (promoters/enhancers), and the colors in this track correspond to their PC1 scores. Furthermore, the promoter of the gene name is centered upon user’s input. The triangles above the regulatory elements represent the referred targets from the available resources in the drop list (<em>GeneHancer</em> interactions by default). The other tracks represent the data intensities, wherein the colors correspond to the normalized epigenetic signals. Additional information are shown as mouse-over tooltips or links in the pop-ups. <a href="https://hdsu-bioquant.github.io/irene-web/docs/Browser.pdf"><img src="https://raw.githubusercontent.com/hdsu-bioquant/irene-web/master/images/Browser.gif" alt="Click to see instructions on how to use this epigenome browser" /></a> <em>Fig. 5 Exploring epigenome browser interface (<a href="https://hdsu-bioquant.github.io/irene-web/docs/Browser.pdf">user guide</a>)</em></p>
</div>
<div id="hallmark-roc" class="section level2">
<h2>Hallmark ROC</h2>
<p>The receiver operating characteristic (<a href="http://hdsu-bioquant.github.io/irene-web/roc.html?sample=CLL">ROC</a>) curves represent the ECDF of marker genes (curated for benchmarking, <a href="https://cancer.sanger.ac.uk/cosmic">COSMIC</a>, <a href="https://www.malacards.org/">MalaCards</a>, <a href="https://www.intogen.org/">IntOGen</a>) of the two rank lists (PromEnh and PromOnly). The curves are updated upon clicking the item listed besides the graph.</p>
<p><a href="https://hdsu-bioquant.github.io/irene-web/docs/ROC.pdf"><img src="https://raw.githubusercontent.com/hdsu-bioquant/irene-web/master/images/ROC.gif" alt="Click to see instructions on how to use this ROC plot" /></a></p>
<p><em>Fig. 6 Exploring ROC plot interface (<a href="https://hdsu-bioquant.github.io/irene-web/docs/ROC.pdf">user guide</a>)</em></p>
</div>
</div>
<div id="case-studies" class="section level1">
<h1>Case studies</h1>
<p>Epigenetic and expression data were downloaded from <a href="http://genboree.org/EdaccData/Release-9/experiment-sample/"><strong>EdaccData Release-9</strong></a>, <a href="http://www.epigenomes.ca/"><strong>CEEHRC</strong></a>, <a href="http://www.blueprint-epigenome.eu/"><strong>BLUEPRINT</strong></a>.</p>
</div>
<div id="acknowledgements" class="section level1">
<h1>Acknowledgements</h1>
<p>The results presented here are in part based upon data generated by The Canadian Epigenetics, Epigenomics, Environment and Health Research Consortium (CEEHRC) initiative funded by the Canadian Institutes of Health Research (CIHR), Genome BC, and Genome Quebec. Information about CEEHRC and the participating investigators and institutions can be found at <a href="http://www.cihr-irsc.gc.ca/e/43734.html" class="uri">http://www.cihr-irsc.gc.ca/e/43734.html</a>.</p>
</div>
<div id="references" class="section level1">
<h1>References</h1>
<ul>
<li><a id="schmitt-ad-2016"></a> Schmitt AD, Hu M, Jung I, et al. A Compendium of Chromatin Contact Maps Reveals Spatially Active Regions in the Human Genome. Cell Reports. 2016;17(8):2042–2059.</li>
<li><a id="ji-h-2013"></a> Ji H, Li X, Wang Qf, et al. Differential principal component analysis of ChIP-seq. Proceedings of the National Academy of Sciences. 2013;110(17):6789–6794.</li>
</ul>
</div>
<div id="faqs" class="section level1">
<h1>FAQs</h1>
<ul>
<li>Q: Package ‘XXX’ is not available (for R version 3.x.x).<br />
A: Please install them manually, using:</li>
</ul>
<pre class="r"><code>source(&quot;https://bioconductor.org/biocLite.R&quot;)
biocLite(&quot;GenomeInfoDb&quot;,&quot;GenomicRanges&quot;,&quot;IRanges&quot;,&quot;preprocessCore&quot;,&quot;S4Vectors&quot;,&quot;XVector&quot;)</code></pre>
<p>For R (version “3.5”) and later, please enter:</p>
<pre class="r"><code>if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)
BiocManager::install(c(&quot;preprocessCore&quot;,&quot;GenomicRanges&quot;), version = &quot;3.8&quot;)</code></pre>
<ul>
<li>Q: Which operation system is <em>Irene</em> able to run on?<br />
A: We have tested <em>Irene</em> under Linux and Mac OSX. For Microsoft Windows, users need to resolve the compiling library themselves, using such as Cygwin or other C libraries.<br />
</li>
<li>Q: How do I open the website for visualizing the outputs?<br />
A: You can run “run.sh” after <em>exportApps</em>. If it does not work, please run it manually in the console, or use the following command:</li>
</ul>
<pre class="sh"><code>python -m SimpleHTTPServer</code></pre>
<p>or</p>
<pre class="sh"><code>python3 -m http.server</code></pre>
<p>, then navigate to <em><a href="http://localhost:8000/" class="uri">http://localhost:8000/</a></em> in your internet browser to see the results.</p>
<div id="license" class="section level2">
<h2># License</h2>
<p>MIT</p>
</div>
</div>
<div id="supplymentary-info" class="section level1">
<h1>Supplymentary Info</h1>
<ul>
<li><a id="bigwigaverageoverbed"></a> Constructing <strong>data</strong> object from BigWig files with <em>bigWigAverageOverBed</em>:</li>
</ul>
<pre class="r"><code>#download genomic coordinates of promoter and enhancers for reading
URL &lt;- &quot;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/promenh.hg19.bed&quot;
BED &lt;- &quot;promenh.hg19.bed&quot;
FUN &lt;- &quot;bigWigAverageOverBed&quot; #or &quot;./bigWigAverageOverBed&quot; if the program is in current folder
download.file(URL, BED)
meta &lt;- data.frame(file=dir(&#39;.&#39;,&#39;*.bigWig$&#39;),group=1,dataset=1,stringsAsFactors=FALSE)
bed  &lt;- read.table(BED,stringsAsFactors=FALSE)
lapply(meta$file,function(f) system(paste0(FUN,&#39; &#39;,f,&#39; &#39;,BED,&#39; &#39;,f,&#39;.out&#39;)))
#use mclapply or other batch scripts if you want to start multiple threads, e.g.
#mclapply(meta$file,function(f) system(paste0(FUN,&#39; &#39;,f,&#39; &#39;,BED,&#39; &#39;,f,&#39;.out&#39;)), mc.cores = 4)
data &lt;- matrix(unlist(lapply(meta$file,function(f)
    read.table(paste0(f,&quot;.out&quot;),stringsAsFactors=FALSE)[,4])), ncol = nrow(meta), byrow = FALSE)
rownames(data) &lt;- read.table(paste0(meta$file[1],&quot;.out&quot;),stringsAsFactors=FALSE)[,1]</code></pre>
<ul>
<li><a id="fullcode">Full R code for all test cases.</li>
</ul>
<pre class="r"><code>library(irene)
options(stringsAsFactors = FALSE)
H1=read.table(&#39;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/PEdistances/H1.hg19.pair&#39;)
H1[,3]=abs(H1[,3]/1e6)
H1[,3]=exp(-20*H1[,3]+1)
data(hprd)
data(markers)

case = &quot;NPC&quot;
load(url(&quot;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/NPC.hg19.rda&quot;))
j = c(1,2,3,5,6)
lbl=c(&#39;Meth&#39;,&#39;K27ac&#39;,&#39;K27me3&#39;,&#39;K36me3&#39;,&#39;K4me1&#39;,&#39;K4me3&#39;,&#39;K9ac&#39;,&#39;K9me3&#39;)
data = data[match(bed[,4],rownames(data)),]
npc = dPCA(meta, bed[i,], data[i,], datasets=j, transform=j, normlen=j, verbose=T)
plotD(npc$Dobs, lbl[j], scales::percent(npc$proj[&#39;percent_var&#39;,]), title=case)
exportD(npc$Dobs, lbl[j], case)
writeData(npc$gr, c(&quot;NPC&quot;, &quot;ESC&quot;), &quot;NPC&quot;)
npc$pg=pageRank(npc$gr, H1, statLog=case, rewire=F)
npc$pg$prom=getPromId(npc$gr)
(npc$auc=plotRank(npc$pg, markers[[case]]))
writeRank(npc$pg[[1]],npc$pg$prom, case)
g=edgeRank(npc$pg[[1]],hprd)
npc$gs=exportMultinets(g, 15, rewire=F)
exportJSONnets(npc$gs, case)
npc$ga = annotNets(npc$gs)
exportJSONpathways(npc$ga, case, n=15)
write.csv(meta,paste0(case,&quot;meta.csv&quot;))
exportApps(case, markers[case])

case = &quot;MSC&quot;
load(url(&quot;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/MSC.hg19.rda&quot;))
j = c(1,2,3,5,6)
lbl=c(&#39;Meth&#39;,&#39;K27ac&#39;,&#39;K27me3&#39;,&#39;K36me3&#39;,&#39;K4me1&#39;,&#39;K4me3&#39;,&#39;K9ac&#39;,&#39;K9me3&#39;)
data = data[match(bed[,4],rownames(data)),]
msc = dPCA(meta, bed[i,], data[i,], datasets=j, transform=j, normlen=j, verbose=T)
plotD(msc$Dobs, lbl[j], scales::percent(msc$proj[&#39;percent_var&#39;,]), title=case)
exportD(msc$Dobs, lbl[j], case)
writeData(msc$gr, c(&quot;MSC&quot;, &quot;ESC&quot;), &quot;MSC&quot;)
msc$pg=pageRank(msc$gr, H1, statLog=case, rewire=F)
msc$pg$prom=getPromId(msc$gr)
(msc$auc=plotRank(msc$pg, markers[[case]]))
writeRank(msc$pg[[1]],msc$pg$prom, case)
g=edgeRank(msc$pg[[1]],hprd)
msc$gs=exportMultinets(g, 15, rewire=F)
exportJSONnets(msc$gs, case)
msc$ga = annotNets(msc$gs)
exportJSONpathways(msc$ga, case, n=15)
write.csv(meta,paste0(case,&quot;meta.csv&quot;))
exportApps(case, markers[case])

case = &quot;MES&quot;
load(url(&quot;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/MES.hg19.rda&quot;))
j = c(1,2,3,5,6)
lbl=c(&#39;Meth&#39;,&#39;K27ac&#39;,&#39;K27me3&#39;,&#39;K36me3&#39;,&#39;K4me1&#39;,&#39;K4me3&#39;,&#39;K9ac&#39;,&#39;K9me3&#39;)
data = data[match(bed[,4],rownames(data)),]
mes = dPCA(meta, bed[i,], data[i,], datasets=j, transform=j, normlen=j, verbose=T)
plotD(mes$Dobs, lbl[j], title=case)
mes$pg=pageRank(mes$gr, H1, rewire=F)
mes$pg$prom=getPromId(mes$gr)

case = &quot;TSC&quot;
load(url(&quot;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/TSC.hg19.rda&quot;))
j = c(1,2,3,5,6)
lbl=c(&#39;Meth&#39;,&#39;K27ac&#39;,&#39;K27me3&#39;,&#39;K36me3&#39;,&#39;K4me1&#39;,&#39;K4me3&#39;,&#39;K9ac&#39;,&#39;K9me3&#39;)
data = data[match(bed[,4],rownames(data)),]
tsc = dPCA(meta, bed[i,], data[i,], datasets=j, transform=j, normlen=j, verbose=T)
plotD(tsc$Dobs, lbl[j], scales::percent(tsc$proj[&#39;percent_var&#39;,]), title=case)
exportD(tsc$Dobs, lbl[j], case)
writeData(tsc$gr, c(&quot;TSC&quot;, &quot;ESC&quot;), &quot;TSC&quot;)
tsc$pg=pageRank(tsc$gr, H1, statLog=case, rewire=F)
tsc$pg$prom=getPromId(tsc$gr)
(tsc$auc=plotRank(tsc$pg, markers[[case]]))
writeRank(tsc$pg[[1]],tsc$pg$prom, case)
g=edgeRank(tsc$pg[[1]],hprd)
tsc$gs=exportMultinets(g, 15, rewire=F)
exportJSONnets(tsc$gs, case)
tsc$ga = annotNets(tsc$gs)
exportJSONpathways(tsc$ga, case, n=15)
write.csv(meta,paste0(case,&quot;meta.csv&quot;))
exportApps(case, markers[case])

case = &quot;CLL&quot;
load(url(&quot;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/CLL.hg19.rda&quot;))
j = c(1,2,6,8)
data = data[match(bed[,4],rownames(data)),]
lbl=c(&#39;K4me1&#39;,&#39;K4me3&#39;,&#39;K9me3&#39;,&#39;K27me3&#39;,&#39;K36me3&#39;,&#39;K27ac&#39;,&#39;Input&#39;,&#39;Meth&#39;)
cll = dPCA(meta, bed[i,], data[i,], datasets=j, transform=j, normlen=j, verbose=T)
plotD(cll$Dobs, lbl[j], scales::percent(cll$proj[&#39;percent_var&#39;,]), title=case)
exportD(cll$Dobs, lbl[j], case)
writeData(cll$gr, c(&quot;CLL&quot;, &quot;Bcell&quot;), &quot;CLL&quot;, intTemp=FALSE)
cll$pg=pageRank(cll$gr, H1, statLog=case, rewire=F)
cll$pg$prom=getPromId(cll$gr)
(cll$auc=plotRank(cll$pg, markers[[case]]))
writeRank(cll$pg[[1]],cll$pg$prom, case)
g=edgeRank(cll$pg[[1]],hprd)
cll$gs=exportMultinets(g, 15, rewire=F)
exportJSONnets(cll$gs, case)
cll$ga = annotNets(cll$gs)
exportJSONpathways(cll$ga, case, n=15)
write.csv(meta,paste0(case,&quot;meta.csv&quot;))
exportApps(case, markers[c(case,&quot;OG&quot;)])

case = &quot;PTC&quot;
load(url(&quot;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/PTC.hg19.rda&quot;))
j = c(1,2,4,6,8)
data = data[match(bed[,4],rownames(data)),]
lbl=c(&#39;K4me1&#39;,&#39;K4me3&#39;,&#39;K9me3&#39;,&#39;K27me3&#39;,&#39;K36me3&#39;,&#39;K27ac&#39;,&#39;Input&#39;,&#39;Meth&#39;)
ptc = dPCA(meta, bed[i,], data[i,], datasets=j, transform=j, normlen=j, verbose=T)
plotD(ptc$Dobs, lbl[j], scales::percent(ptc$proj[&#39;percent_var&#39;,]), title=case)
exportD(ptc$Dobs, lbl[j], case)
writeData(ptc$gr, c(&quot;PTC&quot;, &quot;Thyroid&quot;), &quot;PTC&quot;)
ptc$pg=pageRank(ptc$gr, H1, statLog=case, rewire=F)
ptc$pg$prom=getPromId(ptc$gr)
(ptc$auc=plotRank(ptc$pg, markers[[case]]))
writeRank(ptc$pg[[1]],ptc$pg$prom, case)
g=edgeRank(ptc$pg[[1]],hprd)
ptc$gs=exportMultinets(g, 15, rewire=F)
exportJSONnets(ptc$gs, case)
ptc$ga = annotNets(ptc$gs)
exportJSONpathways(ptc$ga, case, n=15)
write.csv(meta,paste0(case,&quot;meta.csv&quot;))
exportApps(case, markers[c(case,&quot;OG&quot;)])

case = &quot;CRC&quot;
load(url(&quot;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/CRC.hg19.rda&quot;))
j = c(1,2,4,6,8)
data = data[match(bed[,4],rownames(data)),]
lbl=c(&#39;K4me1&#39;,&#39;K4me3&#39;,&#39;K9me3&#39;,&#39;K27me3&#39;,&#39;K36me3&#39;,&#39;K27ac&#39;,&#39;Input&#39;,&#39;Meth&#39;)
crc = dPCA(meta, bed[i,], data[i,], datasets=j, transform=j, normlen=j, verbose=T)
plotD(crc$Dobs, lbl[j], scales::percent(crc$proj[&#39;percent_var&#39;,]), title=case)
exportD(crc$Dobs, lbl[j], case)
writeData(crc$gr, c(&quot;CRC&quot;, &quot;Colon&quot;), &quot;CRC&quot;, intTemp=FALSE)
crc$pg=pageRank(crc$gr, H1, statLog=case, rewire=F)
crc$pg$prom=getPromId(crc$gr)
(crc$auc=plotRank(crc$pg, markers[[case]]))
writeRank(crc$pg[[1]],crc$pg$prom, case)
g=edgeRank(crc$pg[[1]],hprd)
crc$gs=exportMultinets(g, 15, rewire=F)
exportJSONnets(crc$gs, case)
crc$ga = annotNets(crc$gs)
exportJSONpathways(crc$ga, case, n=15)
write.csv(meta,paste0(case,&quot;meta.csv&quot;))
exportApps(case, markers[c(case,&quot;OG&quot;)])

case = &quot;AML&quot;
load(url(&quot;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/nkAML.hg38.rda&quot;))
j = 1:6
data = data[match(bed[,4],rownames(data)),]
lbl=c(&quot;H3K27ac&quot;,&quot;H3K27me3&quot;,&quot;H3K36me3&quot;,&quot;H3K4me1&quot;,&quot;H3K4me3&quot;,&quot;H3K9me3&quot;,&quot;Meth&quot;)
aml = dPCA(meta, bed[i,], data[i,], datasets=j, transform=j, normlen=j, verbose=T)
plotD(aml$Dobs, lbl[j], scales::percent(aml$proj[&#39;percent_var&#39;,]), title=case)
exportD(aml$Dobs, lbl[j], case)
writeData(aml$gr, c(&quot;AML&quot;, &quot;Bcell&quot;), &quot;AML&quot;, intTemp=FALSE)
aml$pg=pageRank(aml$gr, H1, statLog=case, rewire=F)
aml$pg$prom=getPromId(aml$gr)
writeRank(aml$pg[[1]],aml$pg$prom, case)
g=edgeRank(aml$pg[[1]],hprd)
aml$gs=exportMultinets(g, 15, rewire=F)
exportJSONnets(aml$gs, case)
aml$ga = annotNets(aml$gs)
exportJSONpathways(aml$ga, case, n=15)
write.csv(meta,paste0(case,&quot;meta.csv&quot;))
exportApps(case, markers[&quot;OG&quot;])

case = &quot;ALL&quot;
load(url(&quot;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/ALL.hg38.rda&quot;))
j = 1:6
data = data[match(bed[,4],rownames(data)),]
lbl=c(&quot;H3K27ac&quot;,&quot;H3K27me3&quot;,&quot;H3K36me3&quot;,&quot;H3K4me1&quot;,&quot;H3K4me3&quot;,&quot;H3K9me3&quot;,&quot;Meth&quot;)
all = dPCA(meta, bed[i,], data[i,], datasets=j, transform=j, normlen=j, verbose=T)
plotD(all$Dobs, lbl[j], scales::percent(all$proj[&#39;percent_var&#39;,]), title=case)
exportD(all$Dobs, lbl[j], case)
writeData(all$gr, c(&quot;ALL&quot;, &quot;Bcell&quot;), &quot;ALL&quot;, intTemp=FALSE)
all$pg=pageRank(all$gr, H1, statLog=case, rewire=F)
all$pg$prom=getPromId(all$gr)
writeRank(all$pg[[1]],all$pg$prom, case)
g=edgeRank(all$pg[[1]],hprd)
all$gs=exportMultinets(g, 15, rewire=F)
exportJSONnets(all$gs, case)
all$ga = annotNets(all$gs)
exportJSONpathways(all$ga, case, n=15)
write.csv(meta,paste0(case,&quot;meta.csv&quot;))
exportApps(case, markers[&quot;OG&quot;])

case = &quot;mCLL&quot;
load(url(&quot;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/mCLL.hg38.rda&quot;))
j = 1:6
data = data[match(bed[,4],rownames(data)),]
lbl=c(&quot;H3K27ac&quot;,&quot;H3K27me3&quot;,&quot;H3K36me3&quot;,&quot;H3K4me1&quot;,&quot;H3K4me3&quot;,&quot;H3K9me3&quot;,&quot;Meth&quot;)
mcll = dPCA(meta, bed[i,], data[i,], datasets=j, transform=j, normlen=j, verbose=T)
plotD(mcll$Dobs, lbl[j], scales::percent(mcll$proj[&#39;percent_var&#39;,]), title=case)
exportD(mcll$Dobs, lbl[j], case)
writeData(mcll$gr, c(&quot;mCLL&quot;, &quot;Bcell&quot;), &quot;mCLL&quot;, intTemp=FALSE)
mcll$pg=pageRank(mcll$gr, H1, statLog=case, rewire=F)
mcll$pg$prom=getPromId(mcll$gr)
(mcll$auc=plotRank(mcll$pg, markers$CLL))
writeRank(mcll$pg[[1]],mcll$pg$prom, case)
g=edgeRank(mcll$pg[[1]],hprd)
mcll$gs=exportMultinets(g, 15, rewire=F)
exportJSONnets(mcll$gs, case)
mcll$ga = annotNets(mcll$gs)
exportJSONpathways(mcll$ga, case, n=15)
write.csv(meta,paste0(case,&quot;meta.csv&quot;))
exportApps(case, markers[&quot;OG&quot;])

case = &quot;MM&quot;
load(url(&quot;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/MM.hg38.rda&quot;))
j = 1:6
data = data[match(bed[,4],rownames(data)),]
lbl=c(&quot;H3K27ac&quot;,&quot;H3K27me3&quot;,&quot;H3K36me3&quot;,&quot;H3K4me1&quot;,&quot;H3K4me3&quot;,&quot;H3K9me3&quot;,&quot;Meth&quot;)
mm = dPCA(meta, bed[i,], data[i,], datasets=j, transform=j, normlen=j, verbose=T)
plotD(mm$Dobs, lbl[j], scales::percent(mm$proj[&#39;percent_var&#39;,]), title=case)
exportD(mm$Dobs, lbl[j], case)
writeData(mm$gr, c(&quot;MM&quot;, &quot;Bcell&quot;), &quot;MM&quot;, intTemp=FALSE)
mm$pg=pageRank(mm$gr, H1, statLog=case, rewire=F)
mm$pg$prom=getPromId(mm$gr)
writeRank(mm$pg[[1]],mm$pg$prom, case)
g=edgeRank(mm$pg[[1]],hprd)
mm$gs=exportMultinets(g, 15, rewire=F)
exportJSONnets(mm$gs, case)
mm$ga = annotNets(mm$gs)
exportJSONpathways(mm$ga, case, n=15)
write.csv(meta,paste0(case,&quot;meta.csv&quot;))
exportApps(case, markers[&quot;OG&quot;])

case = &quot;MCL&quot;
load(url(&quot;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/MCL.hg38.rda&quot;))
j = 1:6
data = data[match(bed[,4],rownames(data)),]
lbl=c(&quot;H3K27ac&quot;,&quot;H3K27me3&quot;,&quot;H3K36me3&quot;,&quot;H3K4me1&quot;,&quot;H3K4me3&quot;,&quot;H3K9me3&quot;,&quot;Meth&quot;)
mcl = dPCA(meta, bed[i,], data[i,], datasets=j, transform=j, normlen=j, verbose=T)
plotD(mcl$Dobs, lbl[j], scales::percent(mcl$proj[&#39;percent_var&#39;,]), title=case)
exportD(mcl$Dobs, lbl[j], case)
writeData(mcl$gr, c(&quot;MCL&quot;, &quot;Bcell&quot;), &quot;MCL&quot;, intTemp=FALSE)
mcl$pg=pageRank(mcl$gr, H1, statLog=case, rewire=F)
mcl$pg$prom=getPromId(mcl$gr)
writeRank(mcl$pg[[1]],mcl$pg$prom, case)
g=edgeRank(mcl$pg[[1]],hprd)
mcl$gs=exportMultinets(g, 15, rewire=F)
exportJSONnets(mcl$gs, case)
mcl$ga = annotNets(mcl$gs)
exportJSONpathways(mcl$ga, case, n=15)
write.csv(meta,paste0(case,&quot;meta.csv&quot;))
exportApps(case, markers[&quot;OG&quot;])

case = &quot;LGG&quot;
load(url(&quot;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/LGG.hg19.rda&quot;))
j = c(1,2,4,6)
data = data[match(bed[,4],rownames(data)),]
lbl=c(&#39;K4me1&#39;,&#39;K4me3&#39;,&#39;K9me3&#39;,&#39;K27me3&#39;,&#39;K36me3&#39;,&#39;K27ac&#39;,&#39;Input&#39;,&#39;Meth&#39;,&#39;K9ac&#39;)
lgg = dPCA(meta, bed[i,], data[i,], datasets=j, transform=j, normlen=j, nMColMeanCent=1, verbose=T)
plotD(lgg$Dobs, lbl[j], scales::percent(lgg$proj[&#39;percent_var&#39;,]), title=case)
exportD(lgg$Dobs, lbl[j], case)
writeData(lgg$gr, c(&quot;LGG&quot;, &quot;Brain&quot;), &quot;LGG&quot;)
lgg$pg=pageRank(lgg$gr, H1, statLog=case, rewire=F)
lgg$pg$prom=getPromId(lgg$gr)
writeRank(lgg$pg[[1]],lgg$pg$prom, case)
(lgg$auc=plotRank(lgg$pg, markers[[case]]))
g=edgeRank(lgg$pg[[1]],hprd)
lgg$gs=exportMultinets(g, 15, rewire=F)
exportJSONnets(lgg$gs, case)
lgg$ga = annotNets(lgg$gs)
exportJSONpathways(lgg$ga, case, n=15)
write.csv(meta,paste0(case,&quot;meta.csv&quot;))
exportApps(case, markers[c(case,&quot;OG&quot;)])

load(url(&quot;https://raw.githubusercontent.com/hdsu-bioquant/irene-data/master/GSE121719-raw.hg19.rda&quot;))
case = &quot;IDH&quot;
j = 1:6
data = data[match(bed[,4],rownames(data)),]
lbl=c(&#39;K27ac&#39;,&#39;K27me3&#39;,&#39;K4me1&#39;,&#39;K4me3&#39;,&#39;K9me3&#39;,&#39;K36me3&#39;)
glm = dPCA(meta, bed[i,], data[i,], groups=c(1,2), datasets=j, transform=j, normlen=j, verbose=T)
plotD(glm$Dobs, lbl[j], scales::percent(glm$proj[&#39;percent_var&#39;,]), title=case)
exportD(glm$Dobs, lbl[j], case)
writeData(glm$gr, c(case, &quot;MES&quot;), case, intTemp=FALSE)
glm$pg=pageRank(glm$gr, H1, statLog=case, rewire=F)
glm$pg$prom=getPromId(glm$gr)
(glm$auc=plotRank(glm$pg, markers$MGES))
writeRank(glm$pg[[2]],glm$pg$prom, case)
g=edgeRank(glm$pg[[2]],hprd)
glm$gs=exportMultinets(g, 15, rewire=F)
exportJSONnets(glm$gs, case)
glm$ga = annotNets(glm$gs)
exportJSONpathways(glm$ga, case, n=15)
write.csv(meta,paste0(case,&quot;meta.csv&quot;))
exportApps(case, markers[c(&quot;MGES&quot;,&quot;PNGES&quot;)])

case = &quot;RTK_I&quot;
j = 1:6
data = data[match(bed[,4],rownames(data)),]
lbl=c(&#39;K27ac&#39;,&#39;K27me3&#39;,&#39;K4me1&#39;,&#39;K4me3&#39;,&#39;K9me3&#39;,&#39;K36me3&#39;)
glm = dPCA(meta, bed[i,], data[i,], groups=c(3,2), datasets=j, transform=j, normlen=j, verbose=T)
plotD(glm$Dobs, lbl[j], scales::percent(glm$proj[&#39;percent_var&#39;,]), title=case)
exportD(glm$Dobs, lbl[j], case)
writeData(glm$gr, c(case, &quot;MES&quot;), case, intTemp=FALSE)
glm$pg=pageRank(glm$gr, H1, statLog=case, rewire=F)
glm$pg$prom=getPromId(glm$gr)
(glm$auc=plotRank(glm$pg, markers$MGES))
writeRank(glm$pg[[2]],glm$pg$prom, case)
g=edgeRank(glm$pg[[2]],hprd)
glm$gs=exportMultinets(g, 15, rewire=F)
exportJSONnets(glm$gs, case)
glm$ga = annotNets(glm$gs)
exportJSONpathways(glm$ga, case, n=15)
write.csv(meta,paste0(case,&quot;meta.csv&quot;))
exportApps(case, markers[c(&quot;MGES&quot;,&quot;PNGES&quot;)])

case = &quot;RTK_II&quot;
j = 1:6
data = data[match(bed[,4],rownames(data)),]
lbl=c(&#39;K27ac&#39;,&#39;K27me3&#39;,&#39;K4me1&#39;,&#39;K4me3&#39;,&#39;K9me3&#39;,&#39;K36me3&#39;)
glm = dPCA(meta, bed[i,], data[i,], groups=c(4,2), datasets=j, transform=j, normlen=j, verbose=T)
plotD(glm$Dobs, lbl[j], scales::percent(glm$proj[&#39;percent_var&#39;,]), title=case)
exportD(glm$Dobs, lbl[j], case)
writeData(glm$gr, c(case, &quot;MES&quot;), case, intTemp=FALSE)
glm$pg=pageRank(glm$gr, H1, statLog=case, rewire=F)
glm$pg$prom=getPromId(glm$gr)
(glm$auc=plotRank(glm$pg, markers$MGES))
writeRank(glm$pg[[2]],glm$pg$prom, case)
g=edgeRank(glm$pg[[2]],hprd)
glm$gs=exportMultinets(g, 15, rewire=F)
exportJSONnets(glm$gs, case)
glm$ga = annotNets(glm$gs)
exportJSONpathways(glm$ga, case, n=15)
write.csv(meta,paste0(case,&quot;meta.csv&quot;))
exportApps(case, markers[c(&quot;MGES&quot;,&quot;PNGES&quot;)])

case = &quot;PGBM&quot;
j = 1:6
data = data[match(bed[,4],rownames(data)),]
lbl=c(&#39;K27ac&#39;,&#39;K27me3&#39;,&#39;K4me1&#39;,&#39;K4me3&#39;,&#39;K9me3&#39;,&#39;K36me3&#39;)
glm = dPCA(meta, bed[i,], data[i,], groups=list(c(1,3),c(2,4)), datasets=j, transform=j, normlen=j, verbose=F)
plotD(glm$Dobs, lbl[j], scales::percent(glm$proj[&#39;percent_var&#39;,]), title=case)
exportD(glm$Dobs, lbl[j], case)
writeData(glm$gr, c(case, &quot;GBM&quot;), case, intTemp=FALSE)
glm$pg=pageRank(glm$gr, H1, statLog=case, rewire=F)
glm$pg$prom=getPromId(glm$gr)
(glm$auc=plotRank(glm$pg, markers$PNGES))
writeRank(glm$pg[[2]],glm$pg$prom, case)
g=edgeRank(glm$pg[[2]],hprd)
glm$gs=exportMultinets(g, 15, rewire=F)
exportJSONnets(glm$gs, case)
glm$ga = annotNets(glm$gs)
exportJSONpathways(glm$ga, case, n=15)
write.csv(meta,paste0(case,&quot;meta.csv&quot;))
exportApps(case, markers[c(&quot;MGES&quot;,&quot;PNGES&quot;)])</code></pre>
</div>
<div id="session-info" class="section level1">
<h1>Session Info</h1>
<pre class="r"><code>pander(sessionInfo(), compact = FALSE)
</code></pre>
<details>
<p><summary>R version 3.2.2 (2015-08-14):</summary></p>
<p><strong>Platform:</strong> x86_64-pc-linux-gnu (64-bit)</p>
<p><strong>locale:</strong> <em>LC_CTYPE=en_US.UTF-8</em>, <em>LC_NUMERIC=C</em>, <em>LC_TIME=en_US.UTF-8</em>, <em>LC_COLLATE=en_US.UTF-8</em>, <em>LC_MONETARY=en_US.UTF-8</em>, <em>LC_MESSAGES=en_US.UTF-8</em>, <em>LC_PAPER=en_US.UTF-8</em>, <em>LC_NAME=C</em>, <em>LC_ADDRESS=C</em>, <em>LC_TELEPHONE=C</em>, <em>LC_MEASUREMENT=en_US.UTF-8</em> and <em>LC_IDENTIFICATION=C</em></p>
<p><strong>attached base packages:</strong></p>
<ul>
<li>stats4</li>
<li>parallel</li>
<li>stats</li>
<li>graphics</li>
<li>grDevices</li>
<li>utils</li>
<li>datasets</li>
<li>methods</li>
<li>base</li>
</ul>
<p><strong>other attached packages:</strong></p>
<ul>
<li>pander(v.0.6.2)</li>
<li>irene(v.1.0)</li>
<li>enrichR(v.1.0)</li>
<li>stringr(v.1.3.1)</li>
<li>igraph(v.1.2.2)</li>
<li>GenomicRanges(v.1.22.4)</li>
<li>GenomeInfoDb(v.1.6.3)</li>
<li>IRanges(v.2.4.8)</li>
<li>S4Vectors(v.0.8.11)</li>
<li>BiocGenerics(v.0.16.1)</li>
</ul>
<p><strong>loaded via a namespace (and not attached):</strong></p>
<ul>
<li>Rcpp(v.0.12.5)</li>
<li>digest(v.0.6.15)</li>
<li>R6(v.2.2.2)</li>
<li>magrittr(v.1.5)</li>
<li>httr(v.1.3.1)</li>
<li>stringi(v.1.2.4)</li>
<li>zlibbioc(v.1.16.0)</li>
<li>XVector(v.0.10.0)</li>
<li>preprocessCore(v.1.32.0)</li>
<li>rjson(v.0.2.20)</li>
<li>tools(v.3.2.2)</li>
<li>pkgconfig(v.2.0.2)</li>
<li>tcltk(v.3.2.2)</li>
</ul>
</details>
<details>
<p><summary>R version 3.3.3 (2017-03-06):</summary></p>
<p><strong>Platform:</strong> x86_64-apple-darwin13.4.0 (64-bit)</p>
<p><strong>locale:</strong> C||UTF-8||C||C||C||C</p>
<p><strong>attached base packages:</strong></p>
<ul>
<li>parallel</li>
<li>stats4</li>
<li>stats</li>
<li>graphics</li>
<li>grDevices</li>
<li>utils</li>
<li>datasets</li>
<li>methods</li>
<li>base</li>
</ul>
<p><strong>other attached packages:</strong></p>
<ul>
<li>irene(v.1.0)</li>
<li>enrichR(v.1.0)</li>
<li>stringr(v.1.3.1)</li>
<li>igraph(v.1.2.2)</li>
<li>GenomicRanges(v.1.26.4)</li>
<li>GenomeInfoDb(v.1.10.3)</li>
<li>IRanges(v.2.8.2)</li>
<li>S4Vectors(v.0.12.2)</li>
<li>BiocGenerics(v.0.20.0)</li>
<li>pander(v.0.6.1)</li>
</ul>
<p><strong>loaded via a namespace (and not attached):</strong></p>
<ul>
<li>Rcpp(v.0.12.19)</li>
<li>digest(v.0.6.18)</li>
<li>bitops(v.1.0-6)</li>
<li>R6(v.2.3.0)</li>
<li>magrittr(v.1.5)</li>
<li>httr(v.1.3.1)</li>
<li>stringi(v.1.2.4)</li>
<li>zlibbioc(v.1.20.0)</li>
<li>XVector(v.0.14.1)</li>
<li>preprocessCore(v.1.36.0)</li>
<li>rjson(v.0.2.20)</li>
<li>tools(v.3.3.3)</li>
<li>RCurl(v.1.95-4.11)</li>
<li>pkgconfig(v.2.0.2)</li>
</ul>
</details>
<details>
<p><summary>R version 3.5.2 (2018-12-20):</summary></p>
<p><strong>Platform:</strong> x86_64-apple-darwin15.6.0 (64-bit)</p>
<p><strong>locale:</strong> C||UTF-8||C||C||C||C</p>
<p><strong>attached base packages:</strong></p>
<ul>
<li>parallel</li>
<li>stats4</li>
<li>stats</li>
<li>graphics</li>
<li>grDevices</li>
<li>utils</li>
<li>datasets</li>
<li>methods</li>
<li>base</li>
</ul>
<p><strong>other attached packages:</strong></p>
<ul>
<li>pander(v.0.6.3)</li>
<li>irene(v.1.0)</li>
<li>enrichR(v.1.0)</li>
<li>stringr(v.1.3.1)</li>
<li>igraph(v.1.2.2)</li>
<li>GenomicRanges(v.1.34.0)</li>
<li>GenomeInfoDb(v.1.18.1)</li>
<li>IRanges(v.2.16.0)</li>
<li>S4Vectors(v.0.20.1)</li>
<li>BiocGenerics(v.0.28.0)</li>
<li>usethis(v.1.4.0)</li>
<li>devtools(v.2.0.1)</li>
</ul>
<p><strong>loaded via a namespace (and not attached):</strong></p>
<ul>
<li>Rcpp(v.1.0.0)</li>
<li>compiler(v.3.5.2)</li>
<li>BiocManager(v.1.30.4)</li>
<li>XVector(v.0.22.0)</li>
<li>prettyunits(v.1.0.2)</li>
<li>bitops(v.1.0-6)</li>
<li>remotes(v.2.0.2)</li>
<li>tools(v.3.5.2)</li>
<li>zlibbioc(v.1.28.0)</li>
<li>digest(v.0.6.18)</li>
<li>pkgbuild(v.1.0.2)</li>
<li>pkgload(v.1.0.2)</li>
<li>preprocessCore(v.1.44.0)</li>
<li>memoise(v.1.1.0)</li>
<li>pkgconfig(v.2.0.2)</li>
<li>rlang(v.0.3.1)</li>
<li>cli(v.1.0.1)</li>
<li>curl(v.3.3)</li>
<li>GenomeInfoDbData(v.1.2.0)</li>
<li>httr(v.1.4.0)</li>
<li>withr(v.2.1.2)</li>
<li>desc(v.1.2.0)</li>
<li>fs(v.1.2.6)</li>
<li>rprojroot(v.1.3-2)</li>
<li>glue(v.1.3.0)</li>
<li>R6(v.2.3.0)</li>
<li>processx(v.3.2.1)</li>
<li>tcltk(v.3.5.2)</li>
<li>sessioninfo(v.1.1.1)</li>
<li>callr(v.3.1.1)</li>
<li>magrittr(v.1.5)</li>
<li>backports(v.1.1.3)</li>
<li>ps(v.1.3.0)</li>
<li>assertthat(v.0.2.0)</li>
<li>stringi(v.1.2.4)</li>
<li>RCurl(v.1.95-4.11)</li>
<li>rjson(v.0.2.20)</li>
<li>crayon(v.1.3.4)</li>
</ul>
</details>
</div>
<div id="irene" class="section level1">
<h1>Irene</h1>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
